### PR TITLE
Manual pattern matching

### PR DESCRIPTION
**Pull request by xenigmax**
Minimal corrections for the pattern matching part.

This PR is related to #1403  and #955

### CODE REVIEW

File: manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst [hunk 1]
```diff
@@ -17,34 +17,21 @@ Duration
...
+The following line of code shows how the :dox:`Finder` is initialized with an index. In this example, we search for the pattern ``ACGT``.
 
 .. includefrags:: demos/tutorial/indices/base.cpp
       :fragment: finder
 
-After initialization it is possible to use the :dox:`Finder#find` function in order to trigger a search for all occurrences of a given pattern in the underlying :dox:`String` or :dox:`StringSet`.
-In this example, we search for the pattern ``ACGT``:
-
-.. includefrags:: demos/tutorial/indices/base.cpp
-      :fragment: finder2
```
Comment (cpockrandt) [inline manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst]: I think finder2 is not used anywhere else, right? So we can remove the correspoding code from ``demos/tutorial/indices/base.cpp``

File: manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst [hunk 2]
```diff
@@ -17,34 +17,21 @@ Duration
...
-
 Calling the function :dox:`Finder#find` invokes the localization of all occurrences of a given pattern.
 It works by modifying pointers of the ``Finder`` to tables of the index.
-For example, the :dox:`Finder` of ``esaIndex`` stores two pointers, pointing to the first and last suffix array entry that stores an occurrence of the pattern.
-
-The return value of the :dox:`Finder#find` function tells us whether or not a given pattern occurs in the text.
-Furthermore, if there are several instances of a pattern, consecutive calls of :dox:`Finder#find` will modify the :dox:`Finder` such that it points to the next occurrence after each call:
```
Comment (cpockrandt) [inline manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst]: Why did you remove the last sentence? I think it helps to better understand the code below where we have multiple calls.

Comment (xenigmax) [inline manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst]: Ah.. They were move to line 34 (just below). 

But I removed this part since it's already explained in other tutorial. What do you think about this?
> -Furthermore, if there are several instances of a pattern, consecutive calls of :dox:`Finder#find` will modify the :dox:`Finder` such that it points to the next occurrence after each call

Comment (cpockrandt) [inline manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst]: I couldn't find this explanation neither in the related ``online matching tutorial`` nor in the other required tutorials (in a quick way). Sorry if I have missed it. Otherwise I would prefer to keep this sentence :smile:

Comment (xenigmax) [inline manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst]: OK. Let's keep it. !

I thought that below example is just enough.
http://seqan.readthedocs.org/en/feature-manual-restructure/Tutorial/Algorithms/PatternMatching/OnlinePatternMatching.html
> .. the Finder is an iterator that scans over the haystack ..

```
Finder<CharString> finder(haystack);
    Pattern<CharString, Horspool> pattern(needle);
    while (find(finder, pattern))
        std::cout << '[' << beginPosition(finder) << ',' << endPosition(finder) << ")\t" << infix(finder) << std::endl;

    return 0;
}

```

Comment (xenigmax) [inline manual/source/Tutorial/Algorithms/PatternMatching/IndexedPatternMatching.rst]: OK. Let's keep it. !

I thought that below example is just enough.
http://seqan.readthedocs.org/en/feature-manual-restructure/Tutorial/Algorithms/PatternMatching/OnlinePatternMatching.html
> .. the Finder is an iterator that scans over the haystack ..

```
Finder<CharString> finder(haystack);
    Pattern<CharString, Horspool> pattern(needle);
    while (find(finder, pattern))
        std::cout << '[' << beginPosition(finder) << ',' << endPosition(finder) << ")\t" << infix(finder) << std::endl;

    return 0;
}

```
